### PR TITLE
TCGA Classification Analysis Scripts

### PR DESCRIPTION
**Pull request by gwaygenomics**
Here I am adding an analysis of classifying TP53 and RAS status in TCGA samples based on compressed gene expression features.

The `classify.py` script will generate training, testing, and cross validation predictions and performance across all algorithms, z dimensions, and seeds for the two genes. I also add the file `tcga_util.py` that stores helper functions for `classify.py`. I also update the `environment.yml` file.

The next PR will store various visualizations of the results.

### CODE REVIEW

File: 7.tcga-classify/classify.py [hunk 1]
```diff
@@ -0,0 +1,233 @@
...
+        z_matrix_dict[signal][z_dim] = {}
+
+        for z_file in glob.glob("{}/*_z_*".format(matrix_comp_dir)):
+            seed = os.path.basename(z_file).split("_")[1]
+
+            if seed not in z_matrix_dict[signal][z_dim].keys():
+                z_matrix_dict[signal][z_dim][seed] = {}
+
+            if "_test_" in z_file:
+                z_matrix_dict[signal][z_dim][seed]["test"] = z_file
```
Comment (ajlee21) [inline 7.tcga-classify/classify.py]: I haven't seen a data structure like this..is this a hypercube?  Or just a nested dictionary?

File: 7.tcga-classify/classify.py [hunk 2]
```diff
@@ -0,0 +1,233 @@
...
+mut_burden_df = pd.read_table(file, index_col=0)
+
+# What results to track
+full_metrics_list = []
+full_auc_list = []
+full_aupr_list = []
+full_coef_list = []
+all_failed_params = []
+
+# Obtain a list of locations for each feature matrix (X)
```
Comment (ajlee21) [inline 7.tcga-classify/classify.py]: This comment is a bit confusing to me.  Not sure what you mean by "location"  Looks like you're getting your compressed latent space data and populating z_matrix_dict with those values

File: 7.tcga-classify/classify.py [hunk 3]
```diff
@@ -0,0 +1,233 @@
...
+        gene=gene,
+        sample_freeze=sample_freeze_df,
+        mutation_burden=mut_burden_df,
+        filter_count=filter_count,
+        filter_prop=filter_prop,
+        hyper_filter=5,
+    )
+
+    # Now, perform all the analyses for each X matrix
+    for signal in z_matrix_dict.keys():
```
Comment (ajlee21) [inline 7.tcga-classify/classify.py]: Are you testing to see if for each data type (copy number, mutation that is listed at the top) if their compressed form is able to generate the RAS gene expression activity?

File: 7.tcga-classify/classify.py (hunk 1 above)
Comment (gwaygenomics) [reply 7.tcga-classify/classify.py]: nested dictionary. It is how I'm storing specific parameter combinations for the given z matrix

File: 7.tcga-classify/classify.py (hunk 2 above)
Comment (gwaygenomics) [reply 7.tcga-classify/classify.py]: yeah, it is confusing... I will update!

File: 7.tcga-classify/classify.py (hunk 3 above)
Comment (gwaygenomics) [reply 7.tcga-classify/classify.py]: I am testing to see if the compressed gene expression features can be used to predict pathway status in tumors.

E.g. compressing TCGA gene expression into 50 principal components can predict Ras pathway activation in tumors with a 75% AUROC. The predictions are based on if the sample has a copy number gain or non silent mutation in KRAS, HRAS, or NRAS

**Pull request merged**